### ISSUE TITLE
Disable globally progress bar

### ISSUE THREAD

**Issue by rklak** [consolidation/Robo#674]
Is there a way to globally disable the progress bar?

**Comment by rklak** [comment]
Any chance to achieve this?

**Comment by greg-1-anderson** [comment]
I am in favor of improving / disabling the progress bar as desired. When using the logger to output, Robo automatically hides/shows the progress bar around the output, so the stale progress bar image is not left behind. Other code that also outputs, e.g. taskExec et. al., also need to do this operation.

It would also be valuable to have an option to simply disable the progress bar for whatever reason. The workaround for doing this is to set the progress display delay to a very large value. For example:
```
$ ./robo try:progress --progress-delay=99999
➜  Progress bar will display after 99999 seconds of activity.
 [Progress] Processing 1
 [Progress] Processing 2
 [Progress] Processing 3
 [Progress] Processing 4
 [Progress] Processing 5
 [Progress] Processing 6
 [Progress] Processing 7
 [Progress] Processing 8
 [Progress] Processing 9
 [Progress] Processing 10
```
Or do this permanently in your robo.yml:
```
options:
  progress-delay: 99999
```
If Robo implemented EnvConfig (easy -- it just happens not to), then it would also be possible to set an environment variable:
```
ROBO_OPTIONS_PROGRESS_DELAY=99999 ./robo try:progress
```
That at present does not do anything, though.

A pull request to add a separate disable option, or implement EnvConfig, or improve the output of the progress bar would be welcome. I haven't had time to get to these things myself as of yet.

**Comment by jkribeiro** [comment]
Robo 3.0

In my case, even using progress delay option, I'm still having progress output using `taskExec`

```
 [Exec] Running ln -s /efs/testsite/dev/files/private /site/private-files
 [Exec] Done in 0.003s
   19 [============================]
```

**Comment by jkribeiro** [comment]
So, it seems that the progress bar is displayed even for systems that does not support TTY. Here, I'm using alpine linux for an github actions build:

````
Run robo --load-from /root/RoboFile.php --progress-delay=99999 phpcs:coder ${GITHUB_WORKSPACE}
 [Composer\RequireDependency] Requiring packages: /usr/local/bin/composer require drupal/coder --no-interaction
    0 [>---------------------------]

    4 [->--------------------------] [Exec] Running phpcs --standard=Drupal --extensions=inc,install,module,php,profile,test,theme --ignore=\.conf,\.css,\.js,\.md,\.txt,config/*,content/*,core/* --warning-severity=0 --colors /__w/pftest01_profile/pftest01_profile -d date.timezone=UTC
 [Exec] Done in 0.139s


[SUCCESS] Success! No Drupal standards violations found.
    6 [--->------------------------]
````

The progress is messing with the logs. 

@greg-1-anderson I would love to create a PR for this. What is your thought about it, should we create an option to disable the progress globally, or try to determine if the system supports progress or not (not sure how to do this, TBH). Thanks

**Comment by greg-1-anderson** [comment]
Yeah, auto-disabling the progress bar when tty mode is not available is a good idea. You can detect tty mode with [posix_isatty](https://www.php.net/manual/en/function.posix-isatty.php). This function is not available on some systems (e.g. Windows), so you'll need to guard it with a `function_exists`. Off the top of my head, I'm not sure the best place to insert this switch. PRs welcome.

The workaround is to just define `ROBO_OPTIONS_PROGRESS_DELAY` in your CI's variables page.
